lib/game.rb

```diff
@@ -0,0 +1,91 @@
+require 'rspec'
+require './lib/card'
+require './lib/deck'
+require './lib/player'
+require './lib/turn'
+require './lib/game'
+
+class Game 
+  attr_reader :deck, :player1, :player2, :turn
+
+  def initialize 
+    @deck = []
+    @player1 = Player.new("Megan", Deck.new(deck[0..25]))
+    @player2 = Player.new("Aurora", Deck.new(deck[26..51]))
+    @turn = Turn.new(@player1, @player2)
+  end
```

Comment:
I'm going to try and get you unstuck for iteration 3.

Ok so when we call `Game.new` in your runner file, we know that the `initialize` method in your `Game` class is what gets run. In your initialize we set up a `@deck` instance variable to be an empty array (`[]`) and then on the next 2 lines we create two `Player` objects. When you're passing those `Player` objects `deck[0..25]` and `deck[0..25]` let's try and think about what we're actually passing.

What is the value of `deck` currently? It's empty... remember on line 12 we set it to be empty, so it still is empty. So when we are doing `Deck.new(deck[0..25])` or `Deck.new(deck[26..51])` we are passing 2 `Deck` objects with empty cards.

I see that you have a `create_deck` method defined that actually creates the deck of cards and shuffles it but it never gets called. So I would refactor your `initialize` method to look like this:

```ruby
def initialize 
  @deck = []
  create_deck # <== Notice that I add the call to the method that actually creates the deck of cards
  @player1 = Player.new("Megan", Deck.new(deck[0..25])) # <== Now deck[0..25] actually has cards in it
  @player2 = Player.new("Aurora", Deck.new(deck[26..51])) # <== deck[26..51] as well
  # @turn = Turn.new(@player1, @player2)
end
```

You'll notice that I commented out the `@turn` that is because I want to create the turn in the `start` method. Since I know I will need a loop, I want to create a new `turn` object for each.. well -- turn.